helper/resource/testing_new_import_state.go

```diff
@@ -410,16 +434,45 @@ func appendImportBlock(config string, resourceName string, importID string) stri
 		resourceName, importID)
 }
 
+func appendImportBlockWithIdentity(config string, resourceName string, identityValues map[string]any) string {
```

Comment:
Since we are "generating config" here, should we add some tests (like in `helper/resource/importstate/import_block_with_resource_identity_test.go`) for the differing types that we might encounter? We have string attribute encoding covered, so maybe bool, int/float, list of int/float or strings, etc?

Reply:
I like it 💯

Reply:
09e574a3

Reply:
> 09e574a

and subsequent commits 😅